packages/core/contracts-ovm/insured-bridge/implementation/BridgePool.sol

```diff
@@ -326,23 +333,24 @@ contract BridgePool is Testable, BridgePoolInterface, ExpandedERC20 {
         bytes32 depositHash = _getDepositHash(_depositData);
         RelayData storage relay = relays[depositHash];
 
-        // Note `hasPrice` will return false if liveness has not been passed in the optimistic oracle.
-        require(
-            relays[depositHash].relayState == RelayState.Pending &&
-                _getOptimisticOracle().hasPrice(
-                    address(this),
-                    bridgeAdmin.identifier(),
-                    relay.priceRequestTime,
-                    getRelayAncillaryData(_depositData, relay)
-                ),
-            "Relay not settleable"
+        // If relay was disputed, then parties in dispute need to go through OptimisticOracle to resolve dispute.
+        require(relays[depositHash].relayState == RelayState.Pending, "Relay not settleable");
+
+        // Attempt to settle OptimisticOracle price as a convenience for the slow relayer who will receive their
+        // dispute bond back. This doubles as a check that the optimistic price request has resolved because this will
+        // revert if the price has not resolved yet.
+        _getOptimisticOracle().settle(
+            address(this),
+            bridgeAdmin.identifier(),
+            relay.priceRequestTime,
+            getRelayAncillaryData(_depositData, relay)
         );
 
         // Note: Why don't we have to check the value of the price?
```

Comment:
is this comment still valid now? I guess we are not _checking_ for a price but we are _settling. We might want to re-word this to make it clear that we are not checking the outcome of the settlement because in order for this OO request to be sellable the proposal must have _not_ been disputed. If it was disputed.

Reply:
great call, done.